### PR TITLE
Mysql insertMulti() performance improvement

### PR DESCRIPTION
**Pull request by bancer**
See https://github.com/cakephp/cakephp/issues/7856

### CODE REVIEW

File: lib/Cake/Model/Datasource/Database/Mysql.php [hunk 1]
```diff
@@ -849,4 +849,57 @@ protected function _unsigned($real) {
...
+ * @param array $fields The array of field/column names being inserted.
+ * @param array $values The array of values to insert. The values should
+ *   be an array of rows. Each row should have values keyed by the column name.
+ *   Each row must have the values in the same order as $fields.
+ * @return bool
+ */
+	public function insertMulti($table, $fields, $values) {
+		$table = $this->fullTableName($table);
+		$holder = implode(',', array_fill(0, count($fields), '?'));
+		$fields = implode(', ', array_map(array(&$this, 'name'), $fields));
```
Comment (markstory) [inline lib/Cake/Model/Datasource/Database/Mysql.php]: You don't need the `&` anymore.

Comment (markstory) [inline lib/Cake/Model/Datasource/Database/Mysql.php]: Why does one implode have spaces and the other doesn't?

Comment (bancer) [inline lib/Cake/Model/Datasource/Database/Mysql.php]: This part of code is identical to DboSource::insertMulti().

Comment (markstory) [inline lib/Cake/Model/Datasource/Database/Mysql.php]: Ok. I can fix that up then.

File: .gitignore [hunk 2]
```diff
@@ -14,6 +14,9 @@
 #################################
 /nbproject
 .idea
+/.project
+/.buildpath
+/.settings/
```
Comment (ADmad) [inline .gitignore]: Please don't app your project/environment specific ignores to cake. These should go to your environment's global gitignore file.

Comment (thinkingmedia) [inline .gitignore]: http://stackoverflow.com/questions/7335420/global-git-ignore

Comment (markstory) [inline .gitignore]: Aren't these lines for some common editor?

Comment (bancer) [inline .gitignore]: Yes, these are Eclipse paths as noted in the commit comment.

Comment (bancer) [inline .gitignore]: As the comment says "# IDE and editor specific files #" this should be the correct place for them.

Comment (ADmad) [inline .gitignore]: I have always been of the opinion that no IDE specific ignores should be in repo's gitignore. They belong in your global gitignore.

Comment (dereuromark) [inline .gitignore]: Maybe our conventions section in the docs should mention that more clearly?

Comment (bancer) [inline .gitignore]: Should `/nbproject` and `.idea` be removed?

Comment (bancer) [inline .gitignore]: What about  "# OS generated files #" section? Should it be removed?

Comment (dereuromark) [inline .gitignore]: For now just dont add newlines. I think that's all that is necessary at this point.
Keep your PR/Changes as isolated and noise clean as possible always.

Comment (lorenzo) [inline .gitignore]: I personally don't mind adding those lines. They hurt no one and make contribution easier for people to come

**Pull request merged**